### ISSUE TITLE
Anyone have update version about this code.

### ISSUE THREAD

**Issue by asdassZY** [dddoss/tensorflow-socher-ntn#6]
Now, I install tensorflow 1.2.0, but this code used 0.5.0 version. There are a lot of errors when I run this code with my tensorflow version. Does anyone can provide the updated version here. Thanks.

**Comment by kaihuchen** [comment]
I made some changes to this repository to make it run under Tensorflow 1.4. One thing that I found out is 
 that the training process is invariably killed by the system due to the out-of-memory error. It turns out this code's memory usage is very much affected by the number of entities, so I made changes to put a cap on the number of entities used, as well as filtering out those training/testing samples that make reference to 'undefined' entities. This allows the training phase to proceed normally with 20000 entities (out of the originally 38696) on a computer with 16GB RAM.

However, my experiments show that while the training phase can reach pretty good accuracy (up to 97%), running the test samples from the repository invariably got me only around 50% accuracy, which is really no better than random guesses and very far from the 80-90% accuracy as mentioned in the paper. I see two possible explanations to this: one is that I made some mistakes while converting the code, and I am still looking into to. Another possibility is that the system is over-fitting. For this I am testing various sizes for the 'corruption' parameter (which generates random negative examples), but so far I am not seeing improvements yet.  

Will post my version of the code when the problems are fixed. Meanwhile I hope that my experience described above will help someone.
